src/audio.c

```diff
@@ -1181,15 +1181,33 @@ int audio_update(struct audio *a)
 	}
 
 	if (dir & SDP_RECVONLY)
-		err |= aurecv_start_player(a->aur, baresip_auplayl());
+		err |= audio_decoder_set(a, sc->data, sc->pt, sc->rparams);
 
 	if (dir & SDP_SENDONLY)
-		err |= start_source(&a->tx, a, baresip_ausrcl());
+		err |= audio_encoder_set(a, sc->data, sc->pt, sc->params);
+
 	if (err) {
 		warning("audio: start error (%m)\n", err);
 		return err;
 	}
 
+	if (dir & SDP_RECVONLY) {
+		stream_enable_rx(a->strm, true);
+	}
+	else {
+		stream_enable_rx(a->strm, false);
+		aurecv_stop(a->aur);
+	}
+
+	if (dir & SDP_SENDONLY) {
+		err |= start_source(&a->tx, a, baresip_ausrcl());
+		stream_enable_tx(a->strm, true);
+	}
+	else {
+		stream_enable_tx(a->strm, false);
+		stop_tx(&a->tx, a);
+	}
+
```

Comment:
Setting the stream_enable_tx/rx flags looks like it is common code for audio/video ?

Reply:
Yes, and not in correct order in `audio_stop()`.
```
	stop_tx(&a->tx, a);
	stream_enable(a->strm, false);
```

Will try to combine this.

Reply:
- video PR: https://github.com/baresip/baresip/pull/3010